scylla/src/utils/parse.rs

```diff
@@ -87,6 +87,21 @@ impl<'s> ParserState<'s> {
         me
     }
 
+    /// Parses a sequence of digits and '-' as an integer.
+    /// Consumes characters until it finds a character that is not a digit or '-'.
+    ///
+    /// An error is returned if:
+    /// * The first character is not a digit or '-'
+    /// * The integer is larger than i32
+    pub(crate) fn parse_i32(self) -> ParseResult<(i32, Self)> {
+        let (digits, p) = self.take_while(|c| c.is_ascii_digit() || c == '-');
+        if let Ok(value) = digits.parse() {
+            Ok((value, p))
+        } else {
+            Err(p.error(ParseErrorCause::Expected("integer of max length 2**32")))
+        }
```

Comment:
Two minor issues:
- I don't think we expect integer of LENGTH 2**32 ;) The message should be something like "Expected 32-bit signed integer".
- Looking at other usages and Display definition, I think this should use `ParseErrorCause::Other`.

Reply:
Thankyou for the review, I have addressed your feedback.